dbms/src/main/java/org/polypheny/db/transaction/LockManager.java

```diff
@@ -62,27 +66,40 @@ public void lock( LockMode mode, @NonNull Transaction transaction ) throws Deadl
         }
         Thread thread = Thread.currentThread();
 
-        waiters.add( thread );
+        synchronized ( waiters ) {
+            if ( !waiters.add( Triple.of( thread, mode, transaction.getXid() ) ) ){
```

Comment:
According to the [documentation](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/ConcurrentLinkedQueue.html#add(E)), this always returns `true`.